src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -28,7 +30,10 @@ public IEnumerable<int> GetProcesses()
             {
                 //TODO This won't work properly with multi-container scenarios that don't share the process space.
                 //TODO We will need to use DiagnosticsAgent if we are the server.
-                return DiagnosticsClient.GetPublishedProcesses();
+                // Remove the current process from consideration.
+                IList<int> processes = DiagnosticsClient.GetPublishedProcesses().ToList();
+                processes.Remove(Process.GetCurrentProcess().Id);
```

Comment:
Will this work correctly when everything is process 1? Sidecar is also 1 iirc.

Reply:
I've backed out this change and create a separate PR for it.